packages/react/src/components/ComposedModal/ComposedModal.tsx

```diff
@@ -422,12 +433,15 @@ const ComposedModal = React.forwardRef<HTMLDivElement, ComposedModalProps>(
     }, [open, selectorPrimaryFocus, isOpen]);
 
     // Slug is always size `sm`
-    let normalizedSlug;
-    if (slug && slug['type']?.displayName === 'AILabel') {
-      normalizedSlug = React.cloneElement(slug as React.ReactElement<any>, {
-        size: 'sm',
-      });
-    }
+    let normalizedDecorator = useMemo(() => {
+      const element = slug ?? decorator;
+      if (element && element['type']?.displayName === 'AILabel') {
+        return React.cloneElement(element as React.ReactElement<any>, {
+          size: 'sm',
+        });
+      }
+      return isValidElement(element) ? element : null;
+    }, [slug, decorator]);
```

Comment:
i'm wondering if we need to use `useMemo` here
```suggestion
    let normalizedDecorator = React.isValidElement(slug ?? decorator)
        ? (slug ?? decorator)
        : null;
    if (
    normalizedDecorator &&
    normalizedDecorator['type']?.displayName === 'AILabel'
    ) {
    normalizedDecorator = React.cloneElement(
      normalizedDecorator as React.ReactElement<any>,
      {
        size: 'sm',
      }
    );
  }
```

Reply:
I don't think we would need, as decorator isn't something that would change too often.